File: billing-application/src/main/java/com/example/billing/application/command/CreateBudgetAlertRuleCommand.java
package com.example.billing.application.command;

import com.example.billing.domain.shared.Money;

import java.time.Duration;

/** {@code cooldown} null 이면 도메인 default (24h) 사용. */
public record CreateBudgetAlertRuleCommand(
        String idempotencyKey,
        String customerId,
        Money threshold,
        Duration cooldown
) {}

File: billing-application/src/main/kotlin/com/example/billing/application/command/CreateBudgetAlertRuleCommand.kt
package com.example.billing.application.command

import com.example.billing.domain.shared.Money
import java.time.Duration

/** [cooldown] null 이면 도메인 default (24h) 사용. */
@JvmRecord
data class CreateBudgetAlertRuleCommand(
    val idempotencyKey: String,
    val customerId: String,
    val threshold: Money,
    val cooldown: Duration?,
)
